src/hotspot/share/opto/library_call.cpp

```diff
@@ -3769,8 +3769,21 @@ Node* LibraryCallKit::generate_access_flags_guard(Node* kls, int modifier_mask,
 Node* LibraryCallKit::generate_interface_guard(Node* kls, RegionNode* region) {
   return generate_access_flags_guard(kls, JVM_ACC_INTERFACE, 0, region);
 }
+
+// Use this for testing if Klass is_hidden, has_finalizer, and is_cloneable_fast.
+Node* LibraryCallKit::generate_misc_flags_guard(Node* kls, int modifier_mask, int modifier_bits, RegionNode* region) {
```

Comment:
It looks like we could refactor generate_misc_flags_guard and generate_access_flags_guard with a common generate_klass_accessor_guard that takes the offset and type as parameters.

Reply:
Yes, copying this code was a bit painful but it was hard to get right.  I refactored the common code into generate_mods_flags_guard, so we pass in the modifier node and eliminate the duplicate code.